pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -373,3 +429,65 @@ int64_t     *numpy_to_ndarray_shape(int64_t *np_shape, int nd)
     return nd_shape;
 
 }
+
+bool    is_same_shape(t_ndarray dest, t_ndarray src)
+{
+    int i = 0;
+
+    if (dest.nd != src.nd)
+        return false;
+    while (i < dest.nd)
+    {
+        if (dest.shape[i] != src.shape[i])
+            return false;
+        ++i;
+    }
+    return true;
+}
+
+int get_shape_product(int64_t *shape, int nd, int max_nd)
+{
+    int product = 1;
+
+    for (int i = nd; i < max_nd; ++i)
+        product *= shape[i];
+    return (product);
+}
+
+int index_position(t_ndarray arr, int index, int nd)
+{
+    if (arr.order == order_c)
+        return index;
+    if (nd == 0)
+        return (0);
+    if (nd == arr.nd)
+        return (index % arr.shape[nd - 1]) * arr.strides[nd - 1] + index_position(arr, index, nd - 1);
+    int true_index = (index / (get_shape_product(arr.shape, nd, arr.nd)));
+    if (true_index >= arr.shape[nd - 1])
+        true_index = true_index % arr.shape[nd - 1];
+    return (true_index * arr.strides[nd - 1] + index_position(arr, index, nd - 1));
+}
+
+void array_copy_data(t_ndarray dest, t_ndarray src, int64_t offset)
+{
+    unsigned char *d = (unsigned char*)dest.raw_data;
+    unsigned char *s = (unsigned char*)src.raw_data;
+
+    if (dest.order == src.order)
+    {
+        if ((dest.order == order_c) || (dest.order == order_f && is_same_shape(dest, src)))
+        {
+            memcpy(d + offset * dest.type_size, s, src.buffer_size);
+            return ;
+        }
+    }
+
+    int64_t index = 0;

```

Comment:
@aihya This is how it looks like now,  I kept the ifs separated for readability, but maybe it's redundant?